client/state/selectors/get-updateable-jetpack-sites.js

```diff
@@ -6,7 +7,10 @@ import { canJetpackSiteUpdateFiles } from 'calypso/state/sites/selectors';
  * @param   {Object} state Global state tree
  * @returns {Array}        Array of updateable Jetpack sites
  */
-export default function getUpdateableJetpackSites( state ) {
-	const sites = getSelectedOrAllSitesWithPlugins( state );
-	return sites.filter( ( site ) => canJetpackSiteUpdateFiles( state, site.ID ) );
-}
+export default createSelector(
+	( state ) =>
+		getSelectedOrAllSitesWithPlugins( state ).filter( ( site ) =>
+			canJetpackSiteUpdateFiles( state, site.ID )
+		),
+	( state ) => [ state.sites.items ]
```

Comment:
Nice addition, just letting you know that I've faced some issues using array/objects as dependencies, and its usually related to them being recreated every time as a new item. 

One way to avoid it is using data from inside the object as identifier. Example without knowing the `state.sites.items` structure:
```js
( state ) => [ Object.keys(state.sites.items).join('-') ]
```

Not sure if we are facing this issue here, but this may allow us to increase our gains if we were running it more than needed.

Reply:
Interesting, thanks for this additional example @WBerredo!